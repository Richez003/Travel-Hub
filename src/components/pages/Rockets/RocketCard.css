@import url('https://fonts.googleapis.com/css2?family=Roboto+Slab&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@600&family=Roboto+Slab:wght@100&display=swap');

.rocketImage {
  width: 15rem;
}

.rocketDiv {
  display: flex;
  gap: 20px;
  padding: 10px;
}

.rocketBtn {
  width: 8rem;
  padding: 8px;
  color: #fff;
  background: rgb(7, 90, 244);
  outline: 5rem;
  border: none;
  border-radius: 3px;
}

.reserveBtn {
  padding: 5px;
}

.reserveSpan {
  background: rgb(6, 155, 248);
  color: #fff;
  margin-right: 8px;
  border-radius: 5px;
}
