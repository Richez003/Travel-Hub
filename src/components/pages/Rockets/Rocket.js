import { useSelector } from 'react-redux';
import RocketCard from './RocketCard';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);

  return (
    <div>
      <h2>ROCKETS</h2>
      {
        rockets.map((rocket) => (
          <RocketCard key={rocket.id} rocket={rocket} />
        ))
      }
    </div>
  );
};

export default Rockets;
