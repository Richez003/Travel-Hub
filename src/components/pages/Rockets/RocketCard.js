import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveRocket } from '../../../redux/rockets/rocketsReducer';
import './RocketCard.css';

const RocketCard = ({ rocket }) => {
  const dispatch = useDispatch();
  return (
    <div className="rocketDiv">
      <img src={rocket.flickr_images} alt={rocket.rocket_name + rocket.id} className="rocketImage" />
      <div>
        <h3>{rocket.rocket_name}</h3>
        <p className={(rocket.reserved ? 'reserve' : '')}>
          {rocket.reserved && (<span className="reserveSpan">Reserved!</span>)}
          {rocket.description}
        </p>
        <button
          type="button"
          onClick={() => dispatch(reserveRocket(rocket.id))}
          className={(rocket.reserved ? 'reserveBtn' : 'rocketBtn')}
        >
          {(rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket')}
        </button>
      </div>
    </div>
  );
};

RocketCard.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.number,
    rocket_name: PropTypes.string,
    description: PropTypes.string,
    reserved: PropTypes.bool,
    flickr_images: PropTypes.string,
  }).isRequired,
};

export default RocketCard;
