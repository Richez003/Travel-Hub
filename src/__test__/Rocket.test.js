import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/pages/Rockets/Rocket';

describe('Rockets Page Tests', () => {
  test('Rockets Components Renders Correctly in DOM', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(screen.queryByText(/Missions/)).toBeNull();
  });
  it('correctly rendered', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
