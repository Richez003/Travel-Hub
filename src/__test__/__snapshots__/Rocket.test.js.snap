// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Page Tests correctly rendered 1`] = `
<div>
  <h2>
    ROCKETS
  </h2>
</div>
`;
